tests/test_constraint.py

```diff
@@ -147,7 +147,7 @@ def constraint_function_2_dim(x, y):
     params = optimizer.res[0]["params"]
     x, y = params['x'], params['y']
 
-    assert constraint_function_2_dim(x, y) == approx(optimizer.constraint.approx(np.array([x, y])), rel=1e-5, abs=1e-5)
+    assert constraint_function_2_dim(x, y) == approx(optimizer.constraint.approx(np.array([x, y])), rel=1e-3, abs=1e-3)
```

Comment:
I did this adjustment here, do you think it is fine?

Is there any new test I could add that you can think of?

Reply:
> Is there any new test I could add that you can think of?

I think you're good! Coverage is currently increasing with this PR. If you want a closer look, you can see [the report](https://app.codecov.io/gh/fmfn/BayesianOptimization/pull/415/tree/bayes_opt), some lines are only covered partially, but none of is an issue, imo.

Reply:
@leandrobbraga can you confirm that this test fails now?

Reply:
The difference starts in the 4th decimal place

![image](https://github.com/fmfn/BayesianOptimization/assets/18340809/985a1da3-be6a-411e-9df5-167593ea58be)

Reply:
thanks!